counterexamples/homogeneous_prime_not_prime.lean

```diff
@@ -65,16 +65,21 @@ instance : add_comm_monoid two :=
   add_assoc := dec_trivial}
 
 instance : has_lt two :=
-{ lt := λ i j, begin
-  cases i; cases j,
-  { exact false },
-  { exact true },
-  { exact false },
-  { exact false },
-end }
+{ lt := λ i j,
+  match i, j with
+  | z, z := false
+  | z, o := true
+  | o, o := false
+  | o, z := false
+  end }
 
 instance : decidable_rel ((<) : two → two → Prop) :=
-λ i j, by cases i; cases j; apply_instance
+λ i j, match i, j with
+| z, z := decidable.is_false id
+| z, o := decidable.is_true trivial
+| o, o := decidable.is_false id
+| o, z := decidable.is_false id
+end
```

Comment:
I think the 1-liner was fine.

Reply:
I tried `by { cases i; cases j; dec_trivial }` and it doesn't work

Reply:
This works
```lean
λ i j, by { cases i; cases j; try {exact decidable.is_false id}, exact decidable.is_true trivial }
```
but I don't if it better or worse than four lines

Reply:
It worked when I pushed that version earlier; what changed?

Reply:
I changed definition of `lt` using `match end` expression

Reply:
```lean
λ i j, by { cases i; cases j; change decidable true <|> change decidable false; apply_instance, }
```
this also works